<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Editor</class>
 <widget class="QWidget" name="Editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>300</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QPushButton" name="solButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>Size over lifetime</string>
   </property>
  </widget>
  <widget class="QFrame" name="col">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>281</width>
     <height>71</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QCheckBox" name="colEnbaled">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>151</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Color over lifetime</string>
    </property>
   </widget>
   <widget class="QComboBox" name="graType">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>10</y>
      <width>101</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Fixed</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Two-point gradient</string>
     </property>
    </item>
   </widget>
   <widget class="ColorWidget" name="head">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
   <widget class="GradientWidget" name="tail">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>261</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
  </widget>
  <widget class="QToolBox" name="parameterBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>130</y>
     <width>281</width>
     <height>341</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="shape">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>281</width>
      <height>275</height>
     </rect>
    </property>
    <attribute name="label">
     <string>Shape</string>
    </attribute>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>281</width>
       <height>321</height>
      </rect>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::AdjustIgnored</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>279</width>
        <height>319</height>
       </rect>
      </property>
      <widget class="QComboBox" name="typeCombobox">
       <property name="geometry">
        <rect>
         <x>115</x>
         <y>10</y>
         <width>161</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>67</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Type</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>67</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Angle</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Radius</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>150</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Scale</string>
       </property>
      </widget>
      <widget class="QSlider" name="angleSlider">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>50</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>70</y>
         <width>16</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>70</y>
         <width>16</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>90</string>
       </property>
      </widget>
      <widget class="QLabel" name="angleObs">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>50</y>
         <width>31</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="radius">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>90</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="scale">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>140</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="physics">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>281</width>
      <height>275</height>
     </rect>
    </property>
    <attribute name="label">
     <string>Physics</string>
    </attribute>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Gravity modifier</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="gravity">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>0</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="parameters">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>281</width>
      <height>275</height>
     </rect>
    </property>
    <attribute name="label">
     <string>Parameters</string>
    </attribute>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>91</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Start lifetime</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Start speed</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>67</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Start size</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>71</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Max count</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>67</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Rate(/s)</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="life">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>0</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="speed">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>50</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="size">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>100</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QSpinBox" name="max">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>150</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QSpinBox" name="rate">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>200</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>10</y>
     <width>81</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Blur times</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="blurTimes">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>10</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorWidget</class>
   <extends>QFrame</extends>
   <header>colorwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GradientWidget</class>
   <extends>QFrame</extends>
   <header>colorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
